### ISSUE TITLE
Form Validation on Required Fields: Displaying required message too early for some controls

### ISSUE THREAD

**Issue by sproules** [wet-boew/wet-boew#8688]
For file selection (type="file") that is set as required (required="required"), the "The form could not be submitted"/"This field is required" validation messages appears when clicking the Browse (Edge)/Choose File (Chrome) button. Similarly, for dropdown list (select tag) that is set as required, the same validation messages appears when the user switches from a valid option to the "header" option like "Please select an option".  The required validation messages should be displayed once the user submits their information, not as they're trying to enter their information.

Looking at the code in GitHub, the problem appears to be in "formvalid.js" with the following block of code:

/* fixes validation issue (see PR #7913) */
$form.on( "change", "input[type=date], input[type=file], select", function() {
	$form.validate().element( this );
} );

**Comment by thekodester** [comment]
I would argue it not really an issue, as when you click on a file input the browser becomes disabled until you select a file, once you select a file the error is cleared, if you cancel the file request you be informed that a file is required for that form.

**Comment by juleskuehn** [comment]
I agree that the validation is too eager and negatively impacts usability. In my order of preference, validation should occur on:
- submission
- field blur
- field change, with delay

**Comment by EricDunsworth** [comment]
I agree with @juleskuehn.

I feel like the status quo is way too aggressive to the point of making it annoying to fill out a typical form.

I personally prefer the way in which form validation originally worked in WET 4.0. Errors would only appear immediately if a user inserted something that was outright wrong (like an invalid email address) and focused out. Required fields were only validated upon submission.

In other words, users could fill out a form in whatever order they felt like, tab over empty required fields, temporarily change required fields to be empty, etc without randomly triggering required field validation. It was assumed that by the time a user actually tried to submit their form, they'd have filled out all required fields (and only bug them if needed at that point).

Currently, all of the aforementioned situations would trigger required validation in realtime and cause the errors to constantly shift form fields downwards (and upwards once corrected). IMO that makes it really, really annoying to fill out and navigate through WET's forms.

**Comment by juleskuehn** [comment]
Good point about the shifting around @EricDunsworth 
In the current implementation of error messages (without changing to floating call-outs etc) the only solution is to change the behaviour to validate only on form submission.

**Comment by grifare** [comment]
As of now, there are 61 open issues discussing this "form validation" topic! An alternative validation system that would be accepted might close them all. Perhaps we should create a workgroup and resolve this matter once and for all.
There are other problems in addition to the ones listed here: I have a detailed posting under #8091 talking about them, which also incorporates an alternative I coded in 2018.

**Comment by EricDunsworth** [comment]
@duboisp @GormFrank Thoughts? #8456 has also been lingering for over a year.
